base/compiler/abstractinterpretation.jl

```diff
@@ -792,13 +807,14 @@ function is_all_const_arg(argtypes::Vector{Any})
     return true
 end
 
-collect_const_args(arginfo::ArgInfo) = collect_const_args(arginfo.argtypes)
-function collect_const_args(argtypes::Vector{Any})
+collect_const_args((; argtypes)::ArgInfo) = collect_const_args(argtypes)
+function collect_const_args(argtypes::Vector{Any}, start::Int=2)
     return Any[ let a = widenconditional(argtypes[i])
                     isa(a, Const) ? a.val :
                     isconstType(a) ? (a::DataType).parameters[1] :
-                    (a::DataType).instance
-                end for i = 2:length(argtypes) ]
+                    isdefined(a, :instance) ? (a::DataType).instance :
+                    nothing
```

Comment:
Allowing `collect_const_args` to have `nothing` argument seems a bit risky, as we explicitly check that all the arguments can be turned into constant values with `is_all_const_arg` beforehand as for the fully-concrete-eval. Maybe define a separate `collect_semi_const_args` instead?